arrows/mvg/camera_options.h

```diff
@@ -0,0 +1,245 @@
+                                 cam_param_map_t& ext_params,
+                                 std::vector<std::vector<double> >& int_params,
+                                 cam_intrinsic_id_map_t& int_map) const;
+
+  /// update the camera objects using the extracted camera parameters
+  /**
+   *  \param [out] cameras    The map of frame numbers to cameras to update
+   *  \param [in]  ext_params A map from frame number to vector of extrinsic parameters
+   *  \param [in]  int_params A vector of unique camera intrinsic parameter vectors
+   *  \param [in]  int_map    A map from frame number to index into \p int_params.
+   *                          The mapping may be many-to-one for shared intrinsics.
+   *
+   *  The original camera_intrinsic objects are reused if they were not optimized.
+   *  Otherwise new camera_intrinsic instances are created.
+   *
+   *  This function is the inverse of extract_camera_parameters
+   */
+  void
+  update_camera_parameters(camera_map::map_camera_t& cameras,
+                           cam_param_map_t const& ext_params,
+                           std::vector<std::vector<double> > const& int_params,
+                           cam_intrinsic_id_map_t const& int_map) const;
+
+  /// enumerate the intrinsics held constant
+  /**
+   * Based on the settings of the boolean optimization switches
+   * poplulate a vector of indices marking which intrinsic camera
+   * paramaters are held constant.  Indices are:
+   *   - \b 0 : focal length
+   *   - \b 1 : principal point X
+   *   - \b 2 : principal point Y
+   *   - \b 3 : aspect ratio
+   *   - \b 4 : skew
+   *   - \b 5 : radial distortion (k1)
+   *   - \b 6 : radial distortion (k2)
+   *   - \b 7 : tangential distortion (p1)
+   *   - \b 8 : tangential distortion (p2)
+   *   - \b 9 : radial distortion (k3)
+   *   - \b 10 : radial distortion (k4)
+   *   - \b 11 : radial distortion (k5)
+   *   - \b 12 : radial distortion (k6)
+   */
+  std::vector<int> enumerate_constant_intrinsics() const;
+
+  /// option to optimize the focal length
+  bool optimize_focal_length;
+  /// option to optimize aspect ratio
+  bool optimize_aspect_ratio;
+  /// option to optimize principal point
+  bool optimize_principal_point;
+  /// option to optimize skew
+  bool optimize_skew;
+  /// the lens distortion model to use
+  LensDistortionType lens_distortion_type;
+  /// option to optimize radial distortion parameter k1
+  bool optimize_dist_k1;
+  /// option to optimize radial distortion parameter k2
+  bool optimize_dist_k2;
+  /// option to optimize radial distortion parameter k3
+  bool optimize_dist_k3;
+  /// option to optimize tangential distortions parameters p1, p2
+  bool optimize_dist_p1_p2;
+  /// option to optimize radial distortion parameters k4, k5, k6
+  bool optimize_dist_k4_k5_k6;
+  /// the type of sharing of intrinsics between cameras to use
+  CameraIntrinsicShareType camera_intrinsic_share_type;
```

Comment:
The last 4 parameters here are specific to multiple camera optimization or to constraints that we only apply in Ceres.  Let's leave this options in the Ceres solver options subclass.

Reply:
moved them back to ceres